### ISSUE TITLE
DLL Interface Feed Improvements

### ISSUE THREAD

**Issue by Silarn** [EDDiscovery/EDDiscovery#3221]
I realize that support for EDD-EDDM is currently discontinued, but I believe in working on my updated variant I've discovered some problems with the way the DLL Interface feeds data to anything that might want to use it.

After digging into the interface code, it appears that this data is essentially based on EDD's history feed, but it also only captures the data from the previous session + new journal entries processed after starting EDD.

This is problematic due to the way the history list culls certain event types if it thinks they're redundant.

Since the data feed sent over the DLL interface is limited to just a couple log files, it's extremely rare that it ever sends over a full cargo or material list. That means that anything relying on parsing cargo data due to a cargo event (or other similar events) never receives any data, or only partial data when specific items get changed around.

As I see it there are two potential solutions to this. One, when sending the 'current' events, just send the direct journal feed without filtering anything. Two, keep the filtering but add a few 'fake' events in the 'stored' journal file which contain a complete inventory list so that you can try to capture that data during startup.

I've already implemented the direct journal feel locally, which seems to work well enough. I can make a pull request if you'd like.

**Comment by robbyxp1** [comment]
Hi. Thanks for the feedback.

You should be getting a new materials count on every game start, as its at the top of each journal log.   I can't see why you would not. As it replays the last logs thru you should be seeing it.

EDMC only relies itself on reading the last log, to set the state, hence i copied that behaviour.

The DLL sending is after history list processing by design, so the merging removal is part of its behaviour.

So, My plate at the moment is too full to contemplate changing a core part of the program for a program that I'm not currently supporting due to the reasons I specified.  So please don't do a PR, I won't be changing it for this purpose.

Thanks

**Comment by Silarn** [comment]
I don't believe this is the case as it uses EDD's history logs to generate the .edd files. The history logs filter out 'cargo' entries and the like if they don't contain any changes to EDD's internal material storage, which means the complete material list (or even the basic 'empty' cargo event, requiring parsing the cargo.json file etc) never actually gets sent to prompt any materials.

I specifically searched the .edd files for any mention of the items in my cargo and, outside of specific events where I add or remove an item, they never show up.

**Comment by robbyxp1** [comment]
Your right, if its the same, it gets filtered out of the history.  

This system is both in the journal reader, and in the historylist reader, its an intrinsic part of trying to remove all the spam frontier pushed at you.

I'll put this on the action list for consideration.

**Comment by robbyxp1** [comment]
Considering this, the right place to do this is in the EDD EDMC DLL.  It has the materials and commodities list from each journal event pushed thru it at the start.  Save it, and when EDDRefresh comes thru, synthesise a cargo and materials event.

Can't do it inside EDD, does not work, as we only push thru real history entries.

So, its the EDMC DLL which needs updating, if I even restart that.
